### ISSUE TITLE
Slowness while rendering bokeh application

### ISSUE THREAD

**Issue by jundl77** [bokeh/bokeh#9592]
### Problem

I experience significant slowness (delays of up to 10+ sec) when I have 5-6 UI elements in a tab, and starting at as few as 2-3 tabs. **When opening multiple tabs at the same time, bokeh stalls and eventually completely freezes. This is with almost no data.**

Bokeh version: ```1.4.0```

### Example

I have written a small example demonstrating this on my machine (Mac OSX). Given the simplicity of the program, I was not able to reproduce the effect until I had 10 UI elements per tab on 5 tabs.

*FYI*: This does not occur because of the step objects, I have tried the same thing with the more efficient multiline object, and the same thing happens.

```python

#from bokeh.plotting import curdoc
from bokeh.layouts import layout
import bokeh.models as m
from bokeh.models.widgets import Tabs
from bokeh.layouts import column
from bokeh.plotting import figure

import types


class SampleTable:
	def __init__(self):
		self.columns = [
			m.TableColumn(field="1", title="1", width=80),
			m.TableColumn(field="2", title="2", width=80),
			m.TableColumn(field="3", title="3", width=50),
			m.TableColumn(field="4", title="4", width=50),
			m.TableColumn(field="5", title="5", width=80),
			m.TableColumn(field="6", title="6", width=150),
			m.TableColumn(field="7", title="7", width=150),
		]

		self.source = m.ColumnDataSource()
		self.data_table = m.DataTable(source=self.source,
		                              columns=self.columns,
		                              selectable=False,
		                              sortable=False,
		                              index_header='Index',
		                              width=640)
		self.widget = self.data_table

	def get_figure(self):
		return self.widget


class SampleGraph:
	def __init__(self):
		self.p = figure(x_axis_type='datetime',
		                toolbar_location=None,
		                title='Figure')

		self.source = m.ColumnDataSource({
			'timestamp': [1],
			'x0': [1],
			'x1': [1],
			'x2': [1],
			'y0': [1],
			'y1': [1],
			'y2': [1],
		})
		for i in range(3):
			self.p.step(x='timestamp',
			            y=f'x{i}',
			            source=self.source,
			            legend_label="x",
			            line_width=1)

			self.p.step(x='timestamp',
			            y=f'y{i}',
			            source=self.source,
			            legend_label="y",
			            line_width=1)

	def get_figure(self):
		return self.p


class MiniSession:
	def __init__(self):
		self.doc = curdoc()
		self.doc.title = "Sample program"

		self.ui = types.SimpleNamespace()

		self.ui.layout_grid = self._create_layout()
		self.ui.layout = layout([
			self.ui.layout_grid
		], sizing_mode='stretch_both')

		self.doc.add_root(self.ui.layout)

	def _create_layout(self):
		self.ui.tab_content = list(map(lambda i: m.Panel(child=layout(children=self._fill_tab()), title=f"Tab {i}"), range(5)))
		self.ui.tab_bar = Tabs(tabs=self.ui.tab_content)

		return layout(children=[
			[
				self.ui.tab_bar
			]
		])

	def _fill_tab(self):
		return list(map(lambda i: [SampleTable().get_figure(), SampleGraph().get_figure()], range(5)))

session = MiniSession()
```

**Comment by bryevdv** [comment]
Thanks for the complete example @jundl77 FYI I have edited the example to just be a single file rather than directory style app. Also I updated to remove the`widgetbox`. It's not necessary and also is deprecated and will be removed in Bokeh 2.0 in any case. 

cc @mattpap I assume this is just another example of the other recent layout performance issues, in which case we can link them for reference and close this one as a dupe. But I wanted to see if you had any specific comments that might help this particular kind of usage.

**Comment by jundl77** [comment]
Thanks for the quick response (again). 

One thing that crossed my mind is that I saw on other issues/posts that calculating the layout is (or once was) part of the slowness, or has that changed? If so, is it somehow possible to minimize or forgo the layouting system? As in, are there UI elements that require less layouting than others, maybe some that you can literally hard code the size and coordinates into the document?

**Comment by mattpap** [comment]
Duplicate of issue #9320, which was fixed in bokeh 3.0.
